server/odc-service/src/main/java/com/oceanbase/odc/service/task/runtime/RollbackPlanTask.java

```diff
@@ -0,0 +1,276 @@
+        if (StringUtils.isNotBlank(sqlContent)) {
+            this.userInputSqls = SqlUtils.splitWithOffset(this.parameters.getConnectionConfig().getDialectType(),
+                    sqlContent, delimiter);
+        }
+    }
+
+    private void loadUploadFileInputStream() throws IOException {
+        String delimiter = parameters.getDelimiter() != null ? parameters.getDelimiter() : ";";
+        this.uploadFileInputStream = new ByteArrayInputStream(new byte[0]);
+        List<ObjectMetadata> objectMetadataList = this.parameters.getSqlFileObjectMetadatas();
+        if (CollectionUtils.isNotEmpty(objectMetadataList)) {
+            long totalBytes = 0;
+            long maxBytes = parameters.getRollbackProperties().getMaxRollbackContentSizeBytes();
+            for (ObjectMetadata objectMetadata : objectMetadataList) {
+                StorageObject object =
+                        new ObjectStorageHandler(getCloudObjectStorageService(), JobUtils.getExecutorDataPath())
+                                .loadObject(objectMetadata);
+                InputStream current = object.getContent();
+                totalBytes += object.getMetadata().getTotalLength();
+                // remove UTF-8 BOM if exists
+                current.mark(3);
+                byte[] byteSql = new byte[3];
+                if (current.read(byteSql) >= 3 && byteSql[0] == (byte) 0xef && byteSql[1] == (byte) 0xbb
+                        && byteSql[2] == (byte) 0xbf) {
+                    current.reset();
+                    current.skip(3);
+                } else {
+                    current.reset();
+                }
+                if (maxBytes > 0 && totalBytes > maxBytes) {
+                    log.info("The file size is too large and will not be read later, totalSize={} bytes", totalBytes);
+                    break;
+                }
+                this.uploadFileInputStream = new SequenceInputStream(this.uploadFileInputStream, current);
+            }
+            this.uploadFileSqlIterator = SqlUtils.iterator(this.parameters.getConnectionConfig().getDialectType(),
+                    delimiter, this.uploadFileInputStream, StandardCharsets.UTF_8);
+        }
+    }
+
+    private void handleRollbackResult(String rollbackResult) {
+        if (StringUtils.isNotBlank(rollbackResult)) {
+            long totalSize = rollbackResult.getBytes().length;
+            if (totalSize > parameters.getRollbackProperties().getMaxRollbackContentSizeBytes()) {
+                log.warn("Rollback plan result file size exceeds maximum, totalSize={}, taskId={}", totalSize, taskId);
+                throw new UnsupportedSqlTypeForRollbackPlanException("Rollback plan result file size exceeds maximum");
+            }
+            try {
+                String resultFileDownloadUrl = String.format(
+                        "/api/v2/flow/flowInstances/%s/tasks/rollbackPlan/download", parameters.getFlowInstanceId());
```

Comment:
multi same url path can be exacted bo JobUrlConstants

Reply:
This url is used by specific task, no need to put it in `JobUrlConstants`.